sycl/test/regression/cache_test.cpp

```diff
@@ -0,0 +1,148 @@
+    - Create a Queue using a default device selector and verify that the device
+      and platform associated with the queue can be found in the original
+      devices and platforms lists.
+    - Create both a platform and a device object using a native handle, and
+      verify that they are in the original platform and device lists.
+  This test will fail if it tries to create a platform or device that was not
+  in the list of platforms and devices queried in the begininning.
+*/
+
+// Query the device and platform from a queue (constructed with a default device
+// selector) and check if they can be found in the devices and platforms lists
+// that were generated above.
+int queryFromQueue(std::vector<cl::sycl::platform> *platform_list,
+                   std::vector<cl::sycl::device> *device_list) {
+  int failures = 0;
+  cl::sycl::queue Q{cl::sycl::default_selector{}};
+  cl::sycl::device dev = Q.get_info<cl::sycl::info::queue::device>();
+  auto plt = dev.get_platform();
+
+  std::cout << "Platform queried from Queue : "
+            << plt.get_info<cl::sycl::info::platform::name>() << std::endl;
+  auto plt_result =
+      std::find_if(platform_list->begin(), platform_list->end(),
+                   [&](cl::sycl::platform &p) { return p == plt; });
+  if (plt_result != platform_list->end()) {
+    std::cout << "The platform list contains: "
+              << plt.get_info<cl::sycl::info::platform::name>() << std::endl;
+  } else {
+    std::cout << plt.get_info<cl::sycl::info::platform::name>()
+              << " was not in the platform list.\n";
+    failures++;
+  }
+
+  std::cout << "Device queried from Queue : "
+            << plt.get_info<cl::sycl::info::platform::name>() << std::endl;
+  auto dev_result = std::find_if(device_list->begin(), device_list->end(),
+                                 [&](cl::sycl::device &d) { return d == dev; });
+  if (dev_result != device_list->end()) {
+    std::cout << "The device list contains: "
+              << dev.get_info<cl::sycl::info::device::name>() << std::endl;
+  } else {
+    std::cout << dev.get_info<cl::sycl::info::device::name>()
+              << " was not in the device list.\n";
+    failures++;
+  }
+  return failures;
+}
+
+// Create both a platform and a device object using a native handle, and check
+// if those are in the platform and device lists.
+int queryFromNativeHandle(std::vector<cl::sycl::platform> *platform_list,
+                          std::vector<cl::sycl::device> *device_list) {
+  int failures = 0;
+  uint32_t l0_driver_count = 0;
+  zeDriverGet(&l0_driver_count, nullptr);
+  std::vector<ze_driver_handle_t> l0_drivers(l0_driver_count);
+  zeDriverGet(&l0_driver_count, l0_drivers.data());
+
+  uint32_t l0_device_count = 0;
+  zeDeviceGet(l0_drivers[0], &l0_device_count, nullptr);
+  std::vector<ze_device_handle_t> l0_devices(l0_device_count);
+  zeDeviceGet(l0_drivers[0], &l0_device_count, l0_devices.data());
```

Comment:
Could you please add check here that there is at least on driver and one device are available?

Reply:
Done.